src/main/java/Duke.java

```diff
@@ -1,10 +1,260 @@
+                } else if (task instanceof Deadline) {
+                    type = "D";
+                    by = ((Deadline) task).getBy();
+                    fw.write(type + " | " + isDone + " | " + description + " | " + by + "\n");
+                } else if (task instanceof Event) {
+                    type = "E";
+                    start = ((Event) task).getStart();
+                    end = ((Event) task).getEnd();
+                    fw.write(type + " | " + isDone + " | " + description + " | " + start + " | " + end + "\n");
+                }
+            }
+            fw.close();
+        } catch (Exception e) {
+            System.out.println(e);
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Duke duke = new Duke();
+        duke.loadData();
+        duke.sayGreetings();
+
+        while (true) {
+            try {
+                String userInput = duke.sc.nextLine();
+                String[] input = userInput.split(" ", 2);
+                String action = input[0].toUpperCase();
+                System.out.println(LINE);
+                switch (Command.valueOf(action)) {
+                    case BYE:
+                        duke.sayBye();
+                        duke.saveData();
+                        return;
+                    case LIST:
+                        duke.list();
+                        break;
+                    case MARK:
+                        duke.mark(input);
+                        break;
+                    case UNMARK:
+                        duke.unmark(input);
+                        break;
+                    case TODO:
+                        duke.addTodo(input);
+                        break;
+                    case DEADLINE:
+                        duke.addDeadline(input);
+                        break;
+                    case EVENT:
+                        duke.addEvent(input);
+                        break;
+                    case DELETE:
+                        duke.delete(input);
+                        break;
+                    default:
+                        throw new UnknownCommandException();
+                }
+            } catch (IllegalArgumentException e) {
+                if (e.getMessage().contains("No enum constant Duke.Command.")) {
+                    System.out.println("\t " + new UnknownCommandException());
+                } else {
+                    System.out.println("\t" + e.getMessage());
+                }
+            } catch (Exception e) {
+                System.out.println("\t" + e);
+            }
+            System.out.println(LINE);
+        }
     }
 }
```

Comment:
Can consider abstracting into various classes since it can be hard to understand the code as the project expands,
* Duke: Handles the main workflow
* Storage: Handles the saving and loading
* TaskList: Stores the tasks
* Parser: Parses the user input
* Various command classes: Handles the commands given by the user

For the switch statement, `case` clauses should not be indented as specified in the [Coding Standard](https://se-education.org/guides/conventions/java/intermediate.html#layout)